api/src/main/java/org/openmrs/module/mirebalais/apploader/apps/patientregistration/SectionsDefault.java

```diff
@@ -397,8 +404,17 @@ public Question getContactPhoneNumber(Boolean required) {
             f.setFormFieldName("obsgroup.PIH:PATIENT CONTACTS CONSTRUCT.obs.PIH:TELEPHONE NUMBER OF CONTACT");
             f.setLabel("registrationapp.patient.phone.label");
             f.setType("obsgroup");
-            if (required) { f.setCssClasses(Arrays.asList("required")); }
-            f.setWidget(getTextFieldWidget(30));
+			ContactPersonConfigDescriptor contactPersonConfig = config.getRegistrationConfig().getContactPerson();
+            if (contactPersonConfig != null && contactPersonConfig.getRequired() == true) {
+				f.setCssClasses(Arrays.asList("required"));
+			}
+			if(contactPersonConfig != null && contactPersonConfig.getPhoneNumber() != null
+					&& StringUtils.isNotBlank(contactPersonConfig.getPhoneNumber().getRegex())){
+				f.setCssClasses(Arrays.asList("regex"));
+				f.setWidget(getTextFieldWidget(30, contactPersonConfig.getPhoneNumber().getRegex()));
+			} else {
+				f.setWidget(getTextFieldWidget(30));
+			}
```

Comment:
One thing about the regex validation that I find difficult is that there is no guidance in the UI as to what is expected - it enforces a Regex, but doesn't tell you anywhere what the valid format or formats is.

I really think we should add something to the field (probably would work well as a placeholder attribute on the text input).  The placeholder would be human-readable (not the regex itself), i.e. "XXX-XXX-XXXX"

@mogoodrich